### PR TITLE
JBPM-6606: Container redirection test

### PR DESCRIPTION
**Pull request by sutaakar**

### CODE REVIEW

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/AbstractCloudIntegrationTest.java [hunk 1]
```diff
@@ -52,6 +52,7 @@
     protected static final String USER_YODA = "yoda";
 
     protected static final String USERTASK_PROCESS_ID = "definition-project.usertask";
+    protected static final String UPDATED_USERTASK_PROCESS_ID = "definition-project.updated-usertask";
```
Comment (jhrcek) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/AbstractCloudIntegrationTest.java]: What's the difference between USERTASK_PROCESS_ID and UPDATED_USERTASK_PROCESS_ID?

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/Project.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.cloud.integrationtests;
+
+public enum Project {
```
Comment (jakubschwan) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/Project.java]: This naming may cause some conflicts with org.kie.cloud.openshift.resource.Project. Can we name this enum better? e.g. Kjar

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/SmartRouterUpdateRedirection.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    public void testRouterLoadBalancing() {
+        deployProject(kieServerClient1, CONTAINER_ID, Project.DEFINITION_SNAPSHOT);
+        deployProject(kieServerClient2, CONTAINER_ID, Project.DEFINITION_SNAPSHOT);
+
+        verifyProcessAvailableInContainer(smartRouterClient, CONTAINER_ID, USERTASK_PROCESS_ID);
+
+        deployProject(kieServerClient1, CONTAINER_ID_UPDATED, Project.DEFINITION_101_SNAPSHOT);
+
+        for (int i = 0; i < RETRIES_NUMBER; i++) {
+            verifyProcessAvailableInContainer(smartRouterClient, CONTAINER_ID_UPDATED, UPDATED_USERTASK_PROCESS_ID);
```
Comment (jakubschwan) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/SmartRouterUpdateRedirection.java]: We should also verify this for CONTAINER_ALIAS.

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/SmartRouterUpdateRedirection.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    @Test
+    public void testRouterLoadBalancing() {
+        deployProject(kieServerClient1, CONTAINER_ID, Project.DEFINITION_SNAPSHOT);
+        deployProject(kieServerClient2, CONTAINER_ID, Project.DEFINITION_SNAPSHOT);
+
+        verifyProcessAvailableInContainer(smartRouterClient, CONTAINER_ID, USERTASK_PROCESS_ID);
+
+        deployProject(kieServerClient1, CONTAINER_ID_UPDATED, Project.DEFINITION_101_SNAPSHOT);
```
Comment (jakubschwan) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/SmartRouterUpdateRedirection.java]: What if we not deploy a new container, but update the existing one (kieServerClient1.updateReleaseId(CONTAINER_ID, Project.DEFINITION_101_SNAPSHOT)? Can be this valid scenario for Container redirection?

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/SmartRouterUpdateRedirection.java (hunk 3 above)
Comment (sutaakar) [reply test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/SmartRouterUpdateRedirection.java]: Yes, I plan to implement it as second test

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/AbstractCloudIntegrationTest.java (hunk 1 above)
Comment (sutaakar) [reply test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/AbstractCloudIntegrationTest.java]: USERTASK_PROCESS_ID is process id from definition project 1.0.0
UPDATED_USERTASK_PROCESS_ID is process id from definition project 1.0.1

I created it so I am able to verify that the call reached container with definition project 1.0.1

**Comment by sutaakar** [comment]
@jhrcek 
DEFINITION - kjar with version 1.0.0.Final (used for importing to Workbench)
DEFINITION_SNAPSHOT - kjar with version 1.0.0-SNAPSHOT (used for publishing to Maven repo, snapshot version is used to verify that repo will always contain actual artifact)
DEFINITION, DEFINITION_101_SNAPSHOT - kjar with version 1.0.1-SNAPSHOT (used for publishing to Maven repo for updating tests)

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/Project.java (hunk 2 above)
Comment (sutaakar) [reply test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/Project.java]: Good point, will update

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/SmartRouterUpdateRedirection.java (hunk 4 above)
Comment (sutaakar) [reply test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/SmartRouterUpdateRedirection.java]: Sure, for purpose of this test it doesn't matter if new container is installed or old one is updated. I used first thing which came to my mind.

**Comment by sutaakar** [comment]
@jakubschwan Adjusted

**Pull request merged**